### ISSUE TITLE
Project dependency files remain locked indefinitely

### ISSUE THREAD

**Issue by oliversalzburg** [Microsoft/vscode#68222]
Issue Type: <b>Bug</b>

When I open a project in VS Code, VS Code will lock `node_modules` folders in my project dependencies, preventing me from running `npm install` successfully. This worked fine in 1.30.

VS Code version: Code 1.31.0 (7c66f58312b48ed8ca4e387ebd9ffe9605332caa, 2019-02-05T22:35:56.624Z)
OS version: Windows_NT x64 10.0.17763

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-6850K CPU @ 3.60GHz (12 x 3672)|
|GPU Status|2d_canvas: enabled<br>checker_imaging: disabled_off<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>native_gpu_memory_buffers: disabled_software<br>rasterization: enabled<br>surface_synchronization: enabled_on<br>video_decode: enabled<br>webgl: enabled<br>webgl2: enabled|
|Memory (System)|63.91GB (37.85GB free)|
|Process Argv|build-utils-ui/build-utils-ui.code-workspace|
|Screen Reader|no|
|VM|0%|

</details><details><summary>Extensions (13)</summary>

Extension|Author (truncated)|Version
---|---|---
pascal-formatter|ale|2.1.0
better-toml|bun|0.3.2
bracket-pair-colorizer|Coe|1.0.61
vscode-eslint|dba|1.8.0
gitlens|eam|9.5.0
EditorConfig|Edi|0.12.8
JSDocTagComplete|Hoo|0.0.2
vscode-esformatter|how|1.6.0
output-colorizer|IBM|0.1.2
code-beautifier|mic|2.1.0
vscode-lebab|mrm|2.6.0
vscode-puglint|mrm|2.3.0
debugger-for-chrome|msj|4.11.1

(1 theme extensions excluded)

</details>

**Comment by dbaeumer** [comment]
Very likely caused by TypeScript.

**Comment by oliversalzburg** [comment]
I've been through this process a couple of times today and it seems like always the same dependencies are the ones that remain locked.

![image](https://user-images.githubusercontent.com/1658949/52479341-92dd7d80-2ba8-11e9-9610-ee31371f35ae.png)

This is happening a lot, so if there is any information I can gather to help, please let me know.

**Comment by mjbvz** [comment]
Sounds similar to https://github.com/Microsoft/TypeScript/issues/29407

Can you please try disabling all extensions, restarting VS Code to reproduce the problem and then sharing the full process details of the process that is holding the file locks

**Comment by oliversalzburg** [comment]
When I reproduced it right now on a different machine, this process was having locks open:
```
"C:\Users\oliver\AppData\Local\Programs\Microsoft VS Code\Code.exe" "c:\Users\oliver\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js" --useInferredProjectPerProjectRoot --enableTelemetry --cancellationPipeName C:\Users\oliver\AppData\Local\Temp\vscode-typescript\tscancellation-ce678a48cbadcff8df2f.tmp* --locale en --noGetErrOnBackgroundUpdate
```

I'll check again on the original machine tomorrow. I feel like there was something else going on.

**Comment by oliversalzburg** [comment]
Yeah, it always seems to be the `tsserver.js` process, even though the locked directory does not always contain `@types`.
